### PR TITLE
Feature/external tools

### PR DESCRIPTION
**Pull request by trasher**

### CODE REVIEW

File: ajax/dropdownReference.php [hunk 1]
```diff
@@ -65,8 +65,8 @@
                                  '../ajax/dropdownReference.php',
                                  $params);
 
-// Get price
-} elseif (isset($_POST['reference_id'])) {
+   // Get price
```
Comment (orthagh) [inline ajax/dropdownReference.php]: are you sure for the indent of the comment ?

File: front/bill.form.php [hunk 2]
```diff
@@ -73,7 +73,7 @@
                         $ic->update($fields);
                      }
                   }
-                  
+
```
Comment (orthagh) [inline front/bill.form.php]: delete this line ?

File: front/order.form.php [hunk 3]
```diff
@@ -56,20 +56,20 @@
    }
    Html::redirect($url);
 
-/* delete order */
-} elseif (isset($_POST["delete"])) {
+   /* delete order */
```
Comment (orthagh) [inline front/order.form.php]: same for this comment and some below, it seems strange

File: front/reference.form.php [hunk 4]
```diff
@@ -29,16 +29,25 @@
 
 include ("../../../inc/includes.php");
 
-if (!isset($_GET["id"]))            $_GET["id"]           = "";
-if (!isset($_GET["withtemplate"]))  $_GET["withtemplate"] = "";
-if (isset($_POST["popup"]))         $_GET["popup"]        = $_POST["popup"];
-else                                $_GET["popup"]        = "";
-if (!isset($_GET["itemtype"]))      $_GET["itemtype"]     = "";
+if (!isset($_GET["id"])) {
+   $_GET["id"]           = "";
```
Comment (orthagh) [inline front/reference.form.php]: i think we can remove the spaces for the alignement of '='

File: front/reference_supplier.form.php [hunk 5]
```diff
@@ -29,9 +29,15 @@
 
 include ("../../../inc/includes.php");
 
-if(!isset($_GET["id"]))                         $_GET["id"]                         = "";
-if(!isset($_GET["withtemplate"]))               $_GET["withtemplate"]               = "";
-if(!isset($_GET["plugin_order_references_id"])) $_GET["plugin_order_references_id"] = "";
+if (!isset($_GET["id"])) {
+   $_GET["id"]                         = "";
```
Comment (orthagh) [inline front/reference_supplier.form.php]: same for '=' alignement

File: inc/config.class.php [hunk 6]
```diff
@@ -551,68 +553,68 @@ public static function install(Migration $migration) {
...
+         $migration->addField($table, "use_supplier_informations", "tinyint(1) NOT NULL default '1'");
+         $migration->addField($table, "shoudbedelivered_color", "char(20) collate utf8_unicode_ci default '#ff5555'");
+         $migration->addField($table, "copy_documents", "tinyint(1) NOT NULL DEFAULT '0'");
+         $migration->addField($table, "documentcategories_id", "integer");
+         $migration->addField($table, "groups_id_author", "integer");
+         $migration->addField($table, "groups_id_recipient", "integer");
+         $migration->addField($table, "users_id_recipient", "integer");
+
+         $migration->changeField($table, "default_ticketcategories_id",
+                              "default_itilcategories_id", "integer");
```
Comment (orthagh) [inline inc/config.class.php]: This part of the indentation is bad, we need to align the "default_ with the above $table

File: inc/order.class.php [hunk 7]
```diff
@@ -1597,9 +1599,17 @@ public function generateOrder($params) {
             $total_TTC = $prices["priceTTC"] + $postagewithTVA;
 
             if ($signature) {
-               try{$odf->setImage('sign', PLUGIN_ORDER_SIGNATURE_DIR.$signature);} catch(\Odtphp\Exceptions\OdfException $e){}
+               try {
+                  $odf->setImage('sign', PLUGIN_ORDER_SIGNATURE_DIR.$signature);
+               } catch (\Odtphp\Exceptions\OdfException $e) {
+                  $is_cs_happy = true;
```
Comment (orthagh) [inline inc/order.class.php]: $is_cs_happy ?

File: inc/reception.class.php [hunk 8]
```diff
@@ -567,13 +567,13 @@ public function updateReceptionStatus($params) {
...
-         foreach ($params["item"] as $key => $val)
+         foreach ($params["item"] as $key => $val) {
             if ($val == 1) {
                if ($params["itemtype"][$key] == 'SoftwareLicense') {
-                  $this->receptionAllItem($key,$params["plugin_order_references_id"][$key],
-                                          $params["plugin_order_orders_id"],
-                                          $params["delivery_date"], $params["delivery_number"],
-                                          $params["plugin_order_deliverystates_id"]);
+                  $this->receptionAllItem($key, $params["plugin_order_references_id"][$key],
+                                       $params["plugin_order_orders_id"],
```
Comment (orthagh) [inline inc/reception.class.php]: misaligned. Is there a rule who enforce this type of indentation or a miss ?

File: inc/reference_supplier.class.php [hunk 9]
```diff
@@ -360,24 +359,24 @@ public static function install(Migration $migration)
          $migration->addKey($table, "suppliers_id");
          $migration->addKey($table, "plugin_order_references_id");
          $migration->changeField($table, "ID", "id",
-                                 "int(11) NOT NULL auto_increment");
```
Comment (orthagh) [inline inc/reference_supplier.class.php]: misaligned

File: inc/surveysupplier.class.php [hunk 10]
```diff
@@ -442,11 +443,11 @@ public static function install(Migration $migration)
          //1.2.0
          $migration->changeField($table, "ID", "id", "int(11) NOT NULL auto_increment");
          $migration->changeField($table, "FK_order", "plugin_order_orders_id",
-                                 "int(11) NOT NULL default '0' COMMENT 'RELATION to glpi_plugin_order_orders (id)'");
```
Comment (orthagh) [inline inc/surveysupplier.class.php]: same

File: ajax/dropdownReference.php (hunk 1 above)
Comment (trasher) [inline ajax/dropdownReference.php]: Well, our CS implies indentation, so yes, this is required (changes are from phpcbf). I guess this comment is misplaced, it should be inside the else (with its current indentation).

File: front/bill.form.php (hunk 2 above)
Comment (trasher) [inline front/bill.form.php]: Heu.. Ok for this one, but I'm not going to delete all extra lines/comments/whatever on all plugins I've worked on :D

File: inc/config.class.php (hunk 6 above)
Comment (trasher) [inline inc/config.class.php]: We do not have rules for this...

File: front/bill.form.php (hunk 2 above)
Comment (orthagh) [inline front/bill.form.php]: not just this one

File: inc/reception.class.php (hunk 8 above)
Comment (trasher) [inline inc/reception.class.php]: Nope, as I said in a previous comment, we do not have rules for this... :/

File: inc/order.class.php (hunk 7 above)
Comment (trasher) [inline inc/order.class.php]: Required to avoid empty catch

**Pull request merged**